dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImplIntegrationTest.java

```diff
@@ -0,0 +1,309 @@
+package com.dotmarketing.portlets.contentlet.business.web;
+
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.contenttype.transform.JsonTransformer;
+import com.dotcms.datagen.*;
+import com.dotcms.rendering.velocity.viewtools.DotTemplateTool;
+import com.dotcms.repackage.org.directwebremoting.WebContextFactory;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.containers.business.FileAssetContainerUtil;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.containers.model.FileAssetContainer;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+
+import com.dotmarketing.portlets.templates.design.bean.TemplateLayout;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.quartz.QuartzUtils;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.liferay.portal.model.User;
+import com.liferay.util.servlet.SessionMessages;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+import com.dotcms.repackage.org.directwebremoting.WebContext;
+import org.quartz.*;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+public class ContentletWebAPIImplIntegrationTest {
+
+    final String body =
+            "<html>" +
+                    "<head>" +
+                        "#dotParse('%1$s/application/themes/landing-page/html_head.vtl')" +
+                        "<link rel=\"stylesheet\" type=\"text/css\" href=\"/html/css/template/reset-fonts-grids.css\" />" +
+                    "</head>" +
+                    "<body>" +
+                        "<div id=\"resp-template\" name=\"globalContainer\">" +
+                        "<div id=\"hd-template\">" +
+                        "#dotParse('%1$s/application/themes/landing-page/header.vtl')" +
+                        "</div>" +
+                        "<div id=\"bd-template\">" +
+                            "<div id=\"yui-main-template\">" +
+                                "<div class=\"yui-b-template\" id=\"splitBody0\">" +
+                                    "#parseContainer('%1$s/application/containers/default/','1')" +
+                                "</div>" +
+                            "</div>" +
+                        "</div>" +
+                    "</body>" +
+             "</html>";
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5990100)